packages/block-library/src/template-part/index.js

```diff
@@ -39,7 +40,9 @@ export const settings = {
 			return;
 		}
 
-		return startCase( entity.title?.rendered || entity.slug );
+		return (
+			decodeEntities( entity.title?.rendered ) || startCase( entity.slug )
```

Comment:
The `startCase` will remove any entities from titles, so I think only using it for `slug` makes more sense.